### PR TITLE
Fix error when save chain to file and reload(#6889)

### PR DESCRIPTION
**Pull request by sivdead**

### CODE REVIEW

File: langchain/chains/sql_database/base.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def from_llm(
     ) -> SQLDatabaseChain:
         prompt = prompt or SQL_PROMPTS.get(db.dialect, PROMPT)
         llm_chain = LLMChain(llm=llm, prompt=prompt)
-        return cls(llm_chain=llm_chain, database=db, **kwargs)
+        return cls(llm=llm, llm_chain=llm_chain, database=db, **kwargs)
```
Comment (hwchase17) [inline langchain/chains/sql_database/base.py:197]: this should not be passing both, llm is deprecated

File: langchain/llms/openai.py [hunk 2]
```diff
@@ -276,6 +276,8 @@ def _default_params(self) -> Dict[str, Any]:
             "n": self.n,
             "request_timeout": self.request_timeout,
             "logit_bias": self.logit_bias,
+            "openai_api_key": self.openai_api_key,
```
Comment (hwchase17) [inline langchain/llms/openai.py:279]: these should not be passed here but rather elsewhere

**Review by hwchase17 — CHANGES_REQUESTED** [review]

File: langchain/chains/loading.py [hunk 3]
```diff
@@ -352,6 +352,9 @@ def _load_sql_database_chain(config: dict, **kwargs: Any) -> SQLDatabaseChain:
         llm = load_llm_from_config(llm_config)
     elif "llm_path" in config:
         llm = load_llm(config.pop("llm_path"))
+    elif "llm_chain" in config:
```
Comment (hwchase17) [inline langchain/chains/loading.py:355]: i think if llmchain is present, we want to load the whole chain and then pass that in. so NOT pop the llm and use the `from_llm` constructor

Comment (sivdead) [reply langchain/chains/loading.py:355]: ok, i'll refactor this

Comment (sivdead) [reply langchain/chains/loading.py:355]: i notice that when load from string, a secretMap param can be passed, but when load from a file, we don't have this params, is it designed for that? @hwchase17

**Comment by baskaryan** [comment]
stale

**Pull request closed without merging**